pypsa/statistics.py

```diff
@@ -898,48 +980,32 @@ def energy_balance(
             Note that for {'mean', 'sum'} the time series are aggregated to MWh
             using snapshot weightings. With False the time series is given in MW. Defaults to 'sum'.
         """
-        n = self._parent
-
-        @pass_empty_series_if_keyerror
-        def func(n, c):
-            sign = -1 if c in n.branch_components else n.df(c).get("sign", 1)
-            ports = [col[3:] for col in n.df(c).columns if col[:3] == "bus"]
-            p = []
-            for port in ports:
-                mask = n.df(c)[f"bus{port}"] != ""
-                df = sign * n.pnl(c)[f"p{port}"].loc[:, mask]
-                index = get_bus_and_carrier_and_bus_carrier(
-                    n, c, port=port, nice_names=nice_names
-                )
-                index = pd.concat(index, axis=1)[mask]
-                df.columns = pd.MultiIndex.from_frame(index.reindex(df.columns))
-                p.append(df)
-            p = pd.concat(p, axis=1)
-            weights = get_weightings(n, c)
-            return aggregate_timeseries(p, weights, agg=aggregate_time)
-
-        groupby = ["carrier", "bus_carrier"]
-        if not aggregate_bus:
-            groupby.append("bus")
-
-        df = aggregate_components(
-            n,
-            func,
+        if aggregate_bus:
```

Comment:
maybe also add the other attributes from dispatch like `groupby`to the energy_balance and add warning that with `aggregate_bus=True`, this attribute will be ignored